fix(custom-query) : Input gets cleared after saving custom-query

divyanshiGupta opened the pull request:
Fixes https://github.com/openshiftio/openshift.io/issues/3741

Input for custom-query name gets cleared once the query is saved and doesn't show up in the input field of the save custom-query drop-down the next time the save custom-query drop-down is opened.

alien-ike commented:
### Ike Plugins (test-keeper)

Thank you @divyanshiGupta for this contribution!

<img align="left" src="https://cdn.rawgit.com/bartoszmajsak/ike-prow-plugins/2025328b70bd1879520411b3cacadee61a49641a/docs/images/arquillian_ui_failure_128px.png">

It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

Raunak1203 commented:
@divyanshiGupta @nimishamukherjee should we also clear the input on Cancel?

divyanshiGupta commented:
@Raunak1203 I implemented clear input on cancel as well but I noticed that you have mentioned only for save that is why I removed it. I think it should get cleared on cancel as well but let's wait for @nimishamukherjee to reply.

sanbornsen commented on src/app/components_ngrx/toolbar-panel/toolbar-panel.component.html:
```diff
@@ -46,12 +46,12 @@
           <input class="form-control ng-pristine ng-valid ng-touched"
-          (keyup.enter)="saveFilters(filterSaveInp.value)"
-          #filterSaveInp />
+          (keyup.enter)="saveFilters(filterSaveInp,filterSaveInp.value)"
```
Missing space after the `,`

sanbornsen replied:
If we are already passing `filterSaveInp` what is the need of passing a property of it separately?

sanbornsen commented on src/app/components_ngrx/toolbar-panel/toolbar-panel.component.html:
```diff
@@ -46,12 +46,12 @@
-          (keyup.enter)="saveFilters(filterSaveInp.value)"
-          #filterSaveInp />
+          (keyup.enter)="saveFilters(filterSaveInp,filterSaveInp.value)"
+          #filterSaveInp/>
```
Why was the space removed?

sanbornsen commented on src/app/components_ngrx/toolbar-panel/toolbar-panel.component.html:
```diff
@@ -46,12 +46,12 @@
         <div class="save-filter-action">
           <button class="btn btn-primary pull-right margin-left-5 save-cq-btn"
-            (click)="saveFilters(filterSaveInp.value)"
+            (click)="saveFilters(filterSaveInp,filterSaveInp.value)"
```
Same as above comments

sanbornsen commented on src/app/components_ngrx/toolbar-panel/toolbar-panel.component.ts:
```diff
@@ -602,7 +602,7 @@ export class ToolbarPanelComponent implements OnInit, AfterViewInit, OnDestroy {
   }
 
-  saveFilters(filterName: string) {
+  saveFilters(filterSaveInp,filterName: string) {
```
Where is the type?

sanbornsen replied:
Space missing after the `,`

sanbornsen commented on src/app/components_ngrx/toolbar-panel/toolbar-panel.component.ts:
```diff
@@ -617,6 +617,9 @@ export class ToolbarPanelComponent implements OnInit, AfterViewInit, OnDestroy {
       };
       this.store.dispatch(new CustomQueryActions.Add(customQuery));
       this.closeFilterSave();
+      if(filterSaveInp) {
```
Why is the check needed?

divyanshiGupta commented:
@sanbornsen made the changes.

nimishamukherjee commented:
@divyanshiGupta yes please clear on cancel.
